DSharpPlus.Commands/CommandsExtension.cs

```diff
@@ -0,0 +1,166 @@
+    public void AddCommands(IEnumerable<Type> types)
+    {
+        foreach (Type type in types)
+        {
+            if (type.GetCustomAttribute<CommandAttribute>() is not null)
+            {
+                this._commandBuilders.Add(CommandBuilder.From(type));
+                continue;
+            }
+
+            foreach (MethodInfo method in type.GetMethods())
+            {
+                if (method.GetCustomAttribute<CommandAttribute>() is not null)
+                {
+                    this._commandBuilders.Add(CommandBuilder.From(method));
+                }
+            }
+        }
+    }
+
+    public void AddProcessor(ICommandProcessor processor) => this._processors.Add(processor.GetType(), processor);
+    public void AddProcessors(params ICommandProcessor[] processors) => this.AddProcessors((IEnumerable<ICommandProcessor>)processors);
+    public void AddProcessors(IEnumerable<ICommandProcessor> processors)
+    {
+        foreach (ICommandProcessor processor in processors)
+        {
+            this._processors.Add(processor.GetType(), processor);
+        }
+    }
+
+    public TProcessor GetProcessor<TProcessor>() where TProcessor : ICommandProcessor => (TProcessor)this._processors[typeof(TProcessor)];
+
+    public async Task RefreshAsync()
+    {
+        Dictionary<string, Command> commands = [];
+        foreach (CommandBuilder commandBuilder in this._commandBuilders)
+        {
+            try
+            {
+                Command command = commandBuilder.Build();
+                commands.Add(command.Name, command);
+            }
+            catch (Exception error)
+            {
+                this._logger.LogError(error, "Failed to build command '{CommandBuilder}'", commandBuilder.Name);
+            }
+        }
+
+        this.Commands = commands.ToFrozenDictionary();
+
+        foreach (ICommandProcessor processor in this._processors.Values)
+        {
+            await processor.ConfigureAsync(this);
+        }
+    }
+
+    /// <summary>
+    /// The event handler used to log all unhandled exceptions, usually from when <see cref="_commandErrored"/> itself errors.
+    /// </summary>
+    /// <param name="asyncEvent">The event that errored.</param>
+    /// <param name="error">The error that occurred.</param>
+    /// <param name="handler">The handler/method that errored.</param>
+    /// <param name="sender">The extension.</param>
+    /// <param name="eventArgs">The event arguments passed to <paramref name="handler"/>.</param>
+    private static void EverythingWentWrongErrorHandler<TArgs>(AsyncEvent<CommandsExtension, TArgs> asyncEvent, Exception error, AsyncEventHandler<CommandsExtension, TArgs> handler, CommandsExtension sender, TArgs eventArgs) where TArgs : AsyncEventArgs => sender._logger.LogError(error, "Event handler '{Method}' for event {AsyncEvent} threw an unhandled exception.", handler.Method, asyncEvent.Name);
```

Comment:
...this function name could be smaller, but i got zero ideas about new name. CommandErrorHandler? ExtensionErrorHandler?

Reply:
it's fine

Reply:
I mean, it's private. I don't see a reason to rename it